### DISCUSSION TITLE
v9 Alpha1 - fail to init()

### DISCUSSION THREAD

**Discussion by jpotton** [klinecharts/KLineChart#343] [General]
Hey there,

Just becoming familiar with the library so forgive me if these are noob questions. 2 Questions:

1. I'm playing with v9 alpha1, invoking from pure JS and having difficulty creating chart instance. It appears because "data" var is undefined in failing method, the .timestamp attribute therefore doesn't exist and bombs.

![image](https://user-images.githubusercontent.com/9954283/206935581-01833bba-fa5f-4f54-ade9-73470a44727c.png)

2. v8 -> v9 migration notes state I should be importing like so:

> "klinecharts/index.blank' and klinecharts/index.simple are no longer distinguished. Please use import {...} from klinecharts uniformly
https://github.com/liihuu/KLineChart/blob/main/docs/en/v8-to-v9.md

Having plenty of issues with that import method in pure js.  Am I doing something wrong or is that type of invokation only required for Vue/React etc..?

Instead i'm having to use 

`<script src="/static/js/kline_charts/klinecharts.js'></script>`

Thanks in advance,
Jeremy

**Comment by liihuu** [comment]
`<script src="/static/js/kline_charts/klinecharts.js'></script>`
How did get `klinecharts.js`?  
Data must contain some fields. [https://klinecharts.com/guide/data](https://klinecharts.com/guide/data)

You need to do this to introduce through script,
```javascript
klinecharts.init(...)
```

**Reply by jpotton** [reply]
Thanks for taking time to reply...

klinecharts.js came from:

npm install
npm run build
./dist/klinecharts.js

klinecharts.js v9.0.0a1 is served from python:flask

Here is my code (simplified to get to the bottom of this issue).

```
<div id="chart_div"></div>

<script src='/js/klinecharts/klinecharts.js'>
    window.onload = function() { 
        kchart = klinecharts.init('chart_div'); 
    }
<script>
```

This code creates the above exception because data is undefined. Exception occurs before I can call any of these:
* kchart.loadMore(loadmore_cb_func)
* kchart.applyNewData(klines, true)
* kchart.updateData(klines)

In V8 this code works fine.

PS: You've done some excellent work on the library. Very impressed.

**Reply by jpotton** [reply]
@liihuu help please...

**Reply by liihuu** [reply]
I used v9 to do similar operations, but no error occurred. I need more information.

**Reply by jpotton** [reply]
Ok, just to wrap this up, v9.0a4 is working perfectly. Thanks @digitalml for the helpful pm.
